### PR TITLE
VE-47 React UI component improvements

### PR DESCRIPTION
**Pull request by syedsalehinipg**
Closes [VE-47](https://sce.myjetbrains.com/youtrack/issue/VE-47)
Contributes to [VE-33](https://sce.myjetbrains.com/youtrack/issue/VE-33)

## Changes

This PR addresses UI issues mentioned in [#755](https://github.com/your-repo/issues/755)

### TreeViewList
- Removed the gap between each child item.

### TabPanel
- Made the content area scrollable.
- Fixed the tabs' position to the top.

### SurfacePlot
- Added new props: `showGrid`, `labelFontSize`, `exponentFormat`, and `minHeight`.

### LinePlot
- Moved lazy import outside of the export client-only function.
- Added new props: `showGrid`, `labelFontSize`, `exponentFormat`, and `minHeight`.
- Set `markers` and `showGrid` to true by default.
- Set `minHeight` to 400 by default.
- Changed marker color and size to match the existing use case.
- Adjusted margin to match the existing use case.
- Changed grid color to match the existing use case.

## UI/UX

Before:
image

Now:
image

Before:
image

Now:
image

Before:
image

Now:
image

Before:
image

Now:
image

### TreeViewList
- Verify that the gap between each child item is removed.
  - Inspect the spacing visually and ensure that there is no unintended margin or padding between items.

### TabPanel
- Check that when the window is smaller than the content area, a scroll bar appears which only scrolls the content while the tabs remain fixed.
  - Resize the browser window to ensure the scroll bar appears.
  - Ensure that the tabs remain fixed at the top and do not scroll with the content.
  - Verify that the scrolling behavior is smooth and does not impact the usability of the tabs.

### SurfacePlot | LinePlot
- Check that the new props (`showGrid`, `labelFontSize`, `exponentFormat`, and `minHeight`) reflect the relevant changes. Use Storybook for testing these props.
  - `showGrid`: Verify that the grid visibility can be toggled on and off.
  - `labelFontSize`: Change the label font size and confirm the updates reflect accurately on the plot.
  - `exponentFormat`: Check that different exponent formats are applied correctly.
  - `minHeight`: Ensure that the minimum height of the plot is respected and adjust as expected.

### General
- Verify that existing functionality not related to these changes remains unaffected.

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [ ] Branch has been run in docker.
- [x] PR assigned to me or an appropriate delegate.
- [x] Relevant labels added to the PR.
- [ ] Appropriate tests have been added.
- [x] Lint and test workflows pass.

### CODE REVIEW

**Comment by mattcorner** [comment]
In general, I'd say that if we're addressing unrelated feedback on different components, they should be separate PRs.

**Comment by mattcorner** [comment]
The code diff looks weird. Lots of changes (including dependencies) that don't seem to match the list of components mentioned in the PR description.

**Review by mattcorner — CHANGES_REQUESTED** [review]
Code diff doesn't look correct. You've ticked a box to say you reviewed your own code diff. Did you?

**Review by lukemojo — APPROVED** [review]

File: src/SurfacePlot/SurfacePlot.tsx [hunk 1]
```diff
@@ -88,46 +90,46 @@ const SurfacePlot = ({
...
+                exponentformat: exponentFormat,
                 gridcolor:
                   theme.palette.mode === "light"
                     ? ""
                     : theme.palette.grey["500"],
                 title: {
                   font: {
                     family: "Montserrat",
-                    size: 12
+                    size: labelFontSize || undefined
```
[the remark this answers is not in the document]
Comment (syedsalehinipg) [reply src/SurfacePlot/SurfacePlot.tsx:132]: I looked back at the existing Surface3D.js component in Virto (which we moved to React-UI) it seemed a size for the font wasn't supplied originally so I decided to leave it that way but make it a prop in an event where we want to change the font size. The feedback on the review suggested that the size displayed in my PR was too small compared to QA. So I may have to use the size prop when using the component in Virto to make it match.

File: src/LinePlot/LinePlot.tsx
Comment (mattcorner) [inline src/LinePlot/LinePlot.tsx:1]: This file contains several breaking changes from how plots rendered previously. Not sure how they passed review, or why they were made in the first place. These will need to be reverted in a v8.0.1 release @lukemojo as I cannot upgrade in VIRTO.RESULT currently. 

- Plot lines are now the wrong colour
- Plot lines are now the wrong width
- Plot layout has completely changes
- Plot defaults have changed (marked are now on by default).

Comment (mattcorner) [reply src/LinePlot/LinePlot.tsx:1]: [VE-84](https://sce.myjetbrains.com/youtrack/issue/VE-84/Revert-breaking-Plot-changed-in-RUI)

Comment (dmbarry86) [reply src/LinePlot/LinePlot.tsx:1]: > * Plot layout has completely changes

What exactly is meant by this?

**Pull request merged**